google-cloud-clients/google-cloud-core/src/test/java/com/google/cloud/IdentityTest.java

```diff
@@ -141,11 +141,6 @@ public void testValueOfEmpty() {
     Identity.valueOf("");
   }
 
-  @Test(expected = IllegalArgumentException.class)
-  public void testValueOfThreePart() {
-    Identity.valueOf("a:b:c");
```

Comment:
Can you change the test to make sure it takes "a:b" as the identity?

Reply:
> Should we still throw an IllegalArgumentException if there's more parts than we expect?

The output of the split can now no longer be longer than 2, so that's impossible. Now the exception is only thrown if the output is of size zero, which is still correct

Reply:
> Can you change the test to make sure it takes "a:b" as the identity?

For the input string "a:b:c," this test doesn't cause "a" to be the type and "b" to be the value, it causes "a" to be the type and "b:c" to be the value. If you think it's necessary I can add a test for that

Reply:
Let's still have a test to make sure it's split the way we expect